clientd3d/d3drender.h

```diff
@@ -178,10 +178,18 @@ bool ShouldRenderInCurrentPass(bool transparent_pass, bool isTransparent);
 float FovHorizontal(long width);
 float FovVertical(long height);
 
-bool isFogEnabled();
+int getD3dRenderThreshold();
+
 // Returns the max shading range (FINENESS-shade_amount) to FINENESS
 long getShadeAmount();
+
+bool IsManagedTexturesEnabled();
+bool isFogEnabled();
+
 const Vector3D& getSunVector();
 void setWireframeMode(bool isEnabled);
 
+PALETTEENTRY* GetPalette();

```

Comment:
Can these return const references instead?  That would make the call sites simpler.

Also, please add a comment; it's not clear what the difference between a palette and a base paleete is, and why they are different types.